apps/sms/test/marionette/drafts_test.js

```diff
@@ -145,51 +186,240 @@ marionette('Messages Drafts', function() {
         shouldHaveAttachment: true
       };
 
-      messagesStorage = [{
-        id: smsMessage.threadId,
-        body: smsMessage.body,
-        lastMessageType: smsMessage.type,
-        timestamp: smsMessage.timestamp,
-        messages: [smsMessage],
-        participants: [smsMessage.receiver]
-      }];
-
       messagesApp.launch();
-      storage.setMessagesStorage(messagesStorage);
+
+      storage.setMessagesStorage(
+        [conversation], ThreadGenerator.uniqueMessageId
+      );
+
+      inboxView = new InboxView(client);
     });
 
-    test('Draft is correctly saved', function() {
-      messagesApp.Inbox.navigateToComposer();
+    suite('Thread-less drafts', function() {
+      var newMessageView, originalDraftConversation;
 
-      createAndSaveDraft(draft);
+      setup(function() {
+        newMessageView = inboxView.createNewMessage();
+        originalDraftConversation = createAndSaveDraft(newMessageView, draft);
 
-      assertDraft(draft);
+        assertConversation(originalDraftConversation, {
+          isDraft: true,
+          lastMessageType: 'mms',
+          title: draft.recipients[0],
+          bodyText: ''
+        });
```

Comment:
I feel bad doing assertions in the setup; but not too bad :) But if you follow my proposal and merge tests in 1 test scenario, this could be moved there as well.